pkg/crypto/clef/clef.go

```diff
@@ -66,17 +67,39 @@ func DefaultIpcPath() (string, error) {
 	return filepath.Join(home, ".clef", socket), nil
 }
 
+func selectAccount(clef ExternalSignerInterface, ethAddress *common.Address) (accounts.Account, error) {
+	// get the list of available ethereum accounts
+	clefAccounts := clef.Accounts()
+	if len(clefAccounts) == 0 {
+		return accounts.Account{}, ErrNoAccounts
+	}
+
+	if ethAddress == nil {
+		// pick the first account as the one we use
+		return clefAccounts[0], nil
+	}
+
+	for _, availableAccount := range clefAccounts {
+		if availableAccount.Address == *ethAddress {
+			return availableAccount, nil
+		}
+	}
+	return accounts.Account{}, ErrAccountNotAvailable
+}
+
 // NewSigner creates a new connection to the signer at endpoint.
 // As clef does not expose public keys it signs a test message to recover the public key.
```

Comment:
Update the function comment as you are now adding crucial new functionality to this function

Reply:
done